builders/trailsBuilder.js

```diff
@@ -66,6 +66,38 @@ const queries = [
       WHERE (type = 'hike' OR type = 'horse' OR type = 'bike') AND name != ''
     `
   },
+  {
```

Comment:
Question:  Would it be better for us to populate trail start points and end points as part of the migration / build?

Reply:
As I think about this more, a cool thing to do would be to populate something like `priority` in the db which we could assign to different zoom levels. We do a lot of `WHERE ST_Length(geog) > 3000 AND type = 'hike'` and most of it is repetitious. We could assign an integer between one and ten for each trail based on length and type and then we could replace stuff like:

```sql
 SELECT geog::geometry as geom from trails
   WHERE ST_Length(geog) > 3000 AND type = 'hike' OR type = 'horse' OR type = 'bike' OR
   type = 'motorcycle' OR type = 'atv' AND name != ''
   GROUP BY geom;
```

with

```sql
SELECT geog::geometry as geom, priority from trails
  WHERE priority > 2 AND priority < 4
  GROUP BY geom;
```

this wouldn't be a replacement for your idea,

Reply:
actually nvm... using `in` shortens the syntax quite a bit and this is starting to look more and more like a dumb abstraction